Frontend/src/ui/Competition.jsx

```diff
@@ -59,22 +62,114 @@ export default function Competition({ children }) {
                   *Insert Potential organizer announcement or memo for users to
                   view before hitting register*
                 </div>
-                <Button
-                  className={styles.registerButton}
-                  disabled={!competitionInfo['registration_opened?']}
-                  onClick={(_, data) => {
-                    if (!data.disabled) {
-                      if (competitionInfo.use_wca_registration) {
-                        navigate(`${BASE_ROUTE}/${competitionInfo.id}/register`)
-                      } else {
-                        window.location =
-                          competitionInfo.external_registration_page
+                {canAdminCompetitions ? (
+                  <Modal
```

Comment:
This return is already pretty long, it would be good to have a separate component for the admin options.